### PR TITLE
Update cses-1636.mdx

### PR DESCRIPTION
**Pull request by lordboba**
Added Java support for the problem. Note: The MOD Calculation is made slightly faster for addition

_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [ ] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/cses-1636.mdx [hunk 1]
```diff
@@ -77,4 +77,40 @@ int main(){
 
 </CPPSection>
 
+	
+<JavaSection>
+```java
+//lordboba was here
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx:83]: put your name in the author's field instead

File: solutions/cses-1636.mdx [hunk 2]
```diff
@@ -77,4 +77,40 @@ int main(){
...
+		int coins[] = new int[N];
+		for(int i = 0; i < N; ++i) {
+			coins[i] = nextInt();
+		}
+		int MOD = (int)1e9 + 7;
+		for(int j = 0; j < N; ++j) {
+			for(int i = 1; i <= K; ++i) {
+				if(coins[j] > i) {continue;}
+				dp[i] += dp[i-coins[j]];
+				if(dp[i]>MOD) {dp[i]-=MOD;} //slightly faster mod method for addition
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx:106]:
```suggestion
				if (dp[i] > MOD) {
					dp[i]-=MOD;  // slightly faster mod method for addition
				}

File: solutions/cses-1636.mdx [hunk 3]
```diff
@@ -77,4 +77,40 @@ int main(){
...
+	static StreamTokenizer in;
+	static int nextInt() throws IOException{
+		in.nextToken();
+		return (int)in.nval;
+	}
+	public static void main(String[] args) throws IOException{
+		in = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));
+		int N = nextInt(), K = nextInt();
+		int dp[] = new int[K+1];
+		dp[0] = 1;
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx:96]: put this above the for loop where it's actually used

File: solutions/cses-1636.mdx [hunk 4]
```diff
@@ -77,4 +77,40 @@ int main(){
...
+```java
+//lordboba was here
+import java.io.*;
+
+public class CoinComb2 {
+	static StreamTokenizer in;
+	static int nextInt() throws IOException{
+		in.nextToken();
+		return (int)in.nval;
+	}
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx:91]: use bufferedreader, kattio, or scanner

File: solutions/cses-1636.mdx [hunk 5]
```diff
@@ -77,4 +77,40 @@ int main(){
...
+	public static void main(String[] args) throws IOException{
+		in = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));
+		int N = nextInt(), K = nextInt();
+		int dp[] = new int[K+1];
+		dp[0] = 1;
+		int coins[] = new int[N];
+		for(int i = 0; i < N; ++i) {
+			coins[i] = nextInt();
+		}
+		int MOD = (int)1e9 + 7;
```
Comment (SansPapyrus683) [inline solutions/cses-1636.mdx:101]: make this a constant

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
nice first pr

File: solutions/cses-1636.mdx (hunk 4 above)
Comment (nwatx) [reply solutions/cses-1636.mdx:90]: ^

**Comment by nwatx** [comment]
@SansPapyrus683 if you don't mind, can you format this one, otherwise I'll make a pr later

**Comment by nwatx** [comment]
uh oh merge conflict :c

File: solutions/cses-1636.mdx [hunk 6]
```diff
@@ -77,4 +77,41 @@ int main(){
...
+public class CoinComb2 {
+	static StreamTokenizer in;
+	static int nextInt() throws IOException{
+		in.nextToken();
+		return (int)in.nval;
+	}
+	public static void main(String[] args) throws IOException{
+		in = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));
+		int N = nextInt(), K = nextInt();
+		int dp[] = new int[K+1];
```
Comment (nwatx) [inline solutions/cses-1636.mdx:94]:
```suggestion
		int dp[] = new int[K + 1];
```
consistent operator spacing (do for rest as well)

**Review by nwatx — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
please resolve the comments above! other than that, it looks good.

yayy! first contribution :)

**Pull request closed without merging**